buzzmobile/sense/lidar/lidar.py

```diff
@@ -2,96 +2,77 @@
     #generate the image from the cartesian LaserScan points
-    matrix = gen_point_image(lidar_points)
+    lidar_image = gen_point_image(lidar_points)
     #publish the cv image as an imgmsg
-    lidar_publisher.publish(get_lidar_image_message(matrix))
+    lidar_publisher.publish(get_lidar_image_message(lidar_image))
+    
     
-def get_lidar_image_message(matrix):
-    """ Convert an opencv image (matrix) to an imgmsg """
+def get_lidar_image_message(lidar_image):
+    """ Convert an opencv image (image) to an imgmsg """
     try:
-        return bridge.cv2_to_imgmsg(result)
+        color_image = cv2.cvtColor(lidar_image, cv2.COLOR_GRAY2BGR)
+        return bridge.cv2_to_imgmsg(color_image, encoding="bgr8")
     except CvBridgeError as e:
         rospy.loginfo("Error converting lidar image to imgmsg")
 
-# points is a list of tuples. i.e. [(x0, y0), (x1, y1), ...]
+
 def gen_point_image(points):
-    # Generate matrix of all 0's to represent image
-    matrix = np.zeros((c.image_height, c.image_width), np.uint8)
-    matrix.fill(255)
+    """ Generate a cv image from a list of tuple (x, y) coordinates """
+    # Generate image of all 0's to represent image
+    image = np.zeros((image_height, image_width), np.uint8)
+    image.fill(255)
  
-    # Remap points to from lidar coordinates to image coordinates
-    remappedPoints = []
-    for point in points:
-	x = point[0] * c.pixels_per_m
-        y = point[1] * c.pixels_per_m
-	newx = x + (c.image_width / 2)
-        newy = c.image_height - y
-        remappedPoints.append((newx, newy))
-
-    remappedPoints = sorted(remappedPoints, key=lambda xy: xy[0])
-
-    minPoint = min(remappedPoints)
-    maxPoint = max(remappedPoints)
-    
-    remappedPoints.insert(0, (0,0))
-    remappedPoints.insert(1, (0, minPoint[1]))
-    remappedPoints.append((c.image_width, maxPoint[1]))
-    remappedPoints.append((c.image_width, 0))
-
-    #remappedPoints = sorted(remappedPoints, key=lambda xy: xy[0])
-
-    polyPoints = []
-    for pt in remappedPoints:
-        polyPoints.append([pt[0], pt[1]])
+    #get the points at ~pi and ~0
+    min_point = points[0]
+    max_point = points[len(points) - 1]
+ 
+    #add boundary points for line drawing
+    points.insert(0, (0,0))
+    points.insert(1, (0, min_point[1]))
+    points.append((image_width, max_point[1]))
+    points.append((image_width, 0))
     
-    print polyPoints
-
-    #cv2.fillConvexPoly(matrix, np.array(polyPoints, np.int32), 0)
-    cv2.fillPoly(matrix, [np.array(polyPoints, np.int32)], 0)
-    '''cv2.imshow("yo", matrix)
-    cv2.waitKey(0)
-    cv2.destroyAllWindows()'''
-    return matrix
-
-
-#gen_point_image([(1, 2.4), (2, 3.3), (1.5, 9), (3.5, 3.6)])
+    #fill the undrivable portion of the image with black
+    cv2.fillPoly(image, [np.array(points, np.int32)], 0)
+    return image
 
 def lidar_node():
     rospy.init_node('lidar_node', anonymous=True)
     rospy.Subscriber('scan', LaserScan, gen_lidar_image, queue_size=1)
```

Comment:
Correct me if I'm wrong but I believe making a `queue_size=0` is better because it drops any old messages.